docs/rest/FhirPatchRequests.http

```diff
@@ -0,0 +1,281 @@
+# .SUMMARY Sample requests to verify FHIRPath PATCH
+# The assumption for the requests and resources below:
+# The FHIR version is R4 or R5
+
+@hostname = localhost:44348
+
+### Test rest client
+https://{{hostname}}/metadata
+
+### Get the bearer token, if authentication is enabled
+# @name bearer
+POST https://{{hostname}}/connect/token
+content-type: application/x-www-form-urlencoded
+
+grant_type=client_credentials
+&client_id=globalAdminServicePrincipal
+&client_secret=globalAdminServicePrincipal
+&scope=fhir-api
+
+
+### POST one or more Patient resources to test the scenarios below:
+# @name patient
+POST https://{{hostname}}/Patient
+content-type: application/json
+Authorization: Bearer {{bearer.response.body.access_token}}
+
+< .\Data\R4ExamplePatient.json
+
+###
+## Add an address
+
+PATCH https://{{hostname}}/Patient/{{patient.response.body.id}}
+content-type: application/fhir+json
+Authorization: Bearer {{bearer.response.body.access_token}}
+
+{
+    "resourceType": "Parameters",
+    "parameter": [
+        {
+            "name": "operation",
+            "part": [
+                {
+                    "name": "type",
+                    "valueCode": "add"
+                },
+                {
+                    "name": "path",
+                    "valueString": "Patient"
+                },
+                {
+                    "name": "name",
+                    "valueString": "address"
+                },
+                {
+                    "name": "value",
+                    "valueAddress": {
+                        "use": "home",
+                        "line": [
+                            "23 thule st",
+                            "avon"
+                        ],
+                        "city": "Big Smoke",
+                        "country": "erewhon",
+                        "text": "23 thule st"
+                    }
+                },
+            ]
+        }
+    ]
+}
+
+###
+## Remove a phone number
+
+PATCH https://{{hostname}}/Patient/{{patient.response.body.id}}
+content-type: application/fhir+json
+Authorization: Bearer {{bearer.response.body.access_token}}
+
+{
+    "resourceType": "Parameters",
+    "parameter": [
+        {
+            "name": "operation",
+            "part": [
+                {
+                    "name": "type",
+                    "valueCode": "delete"
+                },
+                {
+                    "name": "path",
+                    "valueString": "Patient.telecom.where(use = 'home')"
+                }
+            ]
+        }
+    ]
+}
+
+###
+## Add a phone number
```

Comment:
Super nitpick: I think this example should be above the example that removes a phone number to allow the remove example to demonstrate the functionality of the condition in the valueString. Right now there is only one phone number so it is unclear if the conditional statement is working.

Reply:
Good callout. I'm going to run all the rest requests again in a couple of hours for a sanity check.

Reply:
All looks good on these @LTA-Thinking. Tested and still working as intended.